src/basilisp/lang/obj.py

```diff
@@ -76,17 +77,46 @@ def map_lrepr(
     with the start and end string supplied. The entries argument must be a
     callable which will produce tuples of key-value pairs.
 
+    If the keyword argument print_namespace_maps is True and all keys
+    are symbols or keywords and they share the same namespace, then
+    print the namespace of the keys at the beginning of the map
+    instead of beside the keys.
+
     The keyword arguments will be passed along to lrepr for the sequence
-    elements."""
+    elements.
+
+    """
+    from basilisp.lang import keyword as kw  # pylint: disable=cyclic-import
+    from basilisp.lang import symbol as sym  # pylint: disable=cyclic-import
+
     print_level = kwargs["print_level"]
     if isinstance(print_level, int) and print_level < 1:
         return SURPASSED_PRINT_LEVEL
 
     kwargs = _process_kwargs(**kwargs)
 
+    def check_same_ns():
+        nses = set()
+        for k, _ in entries():
```

Comment:
Can you just double check that calling `entries` twice is ok? I think it calls `.items()` which should be fine, but just want to verify we're not exhausting the collection by iterating over it twice.

Reply:
I can confirm the two instances calling this function are passing in `items`

``` python
@lrepr.register(dict)
def _lrepr_py_dict(o: dict, **kwargs: Unpack[PrintSettings]) -> str:
    return f"#py {map_lrepr(o.items, '{', '}', **kwargs)}"

# and
def _lrepr(self, **kwargs: Unpack[PrintSettings]):
    return map_lrepr(
        self._inner.items, start="{", end="}", meta=self._meta, **kwargs
        )
```